hoomd/hpmc/integrate.py

```diff
@@ -208,7 +209,7 @@ def type_shapes(self):
 
     def _return_type_shapes(self):
         if not self._attached:
-            return None
+            raise DataAccessError("type_shapes")
```

Comment:
Use `log(requires_attach=True)` on the `type_shapes` loggable quantity for subclasses.